src/util/StringUtils.h

```diff
@@ -244,6 +260,35 @@ constexpr bool constantTimeEquals(std::string_view view1,
   return impl(toVolatile(view1), toVolatile(view2));
 }
 
+// _________________________________________________________________________
+template <typename Range, typename TranslationFunction>
+static std::string listToString(Range&& r,
+                                TranslationFunction translationFunction,
+                                std::string_view listItemSeparator) {
+  if (r.empty()) {
+    return "";
+  }
+
+  std::ostringstream stream;
+
+  /*
+  TODO<C++23>: This can be a combination of `std::views::transform` and
+  `std::views::join_with`. After that, we just have to insert all the elements
+  of the new view into the stream.
+  */
+  forEachExcludingTheLastOne(
+      AD_FWD(r),
```

Comment:
We now have `views::transform`, so use can use `r | std::views::transform(AD_FWD(translationFunction))` as the first argument, and make the two lambdas much simpler (they don't need to stransform anymore).

Reply:
Question: Whats the point in forwarding a function? I thought forwarding existed, so that we could use move semantics for r-values?

Reply:
Additionally: Is there a reason, we I should not also use `std::views::join_with`?

Reply:
I did it a bit differently: [1f58677](https://github.com/ad-freiburg/qlever/pull/1010/commits/1f586776b8fe3f808dc42ecb7997f2a9e9f6f63a)
Is that okay?

Reply:
If you meant that you used the "normal" function call syntax for `transform` isntead of the `|` operatyor, and bount
the result to an explicit variable, that is ddefinitely okay. (I like the extra line here).